index.js

```diff
@@ -25,6 +25,17 @@ const register = (server, pluginOptions) => {
 
     if (typeof options.validateKey !== 'function') {
       validateKey = (token) => {
+        // if the apiKeys are listed in an array, see if any match:
+        if (Array.isArray(options.apiKeys)) {
+          const key = options.apiKeys.find(entry => entry.key === token);
+          if (key) {
+            // don't need the key name in the credentials:
+            delete key.key;
+            return { isValid: true, credentials: key };
```

Comment:
I think credentials should still be an object.

To maintain compatibility probably something like:

```js
{ server: true }
```